spyder/api/plugins.py

```diff
@@ -8,252 +8,51 @@
-        are not captured and are propagated return False.
-        """
-        event_type = event.type()
-        if event_type == QEvent.MouseButtonPress:
-            self.tab_pressed(event)
-            return False
-        if event_type == QEvent.MouseMove:
-            self.tab_moved(event)
-            return True
-        if event_type == QEvent.MouseButtonRelease:
-            self.tab_released(event)
-            return True
-        return False
-
-    def tab_pressed(self, event):
-        """Method called when a tab from a QTabBar has been pressed."""
-        self.from_index = self.dock_tabbar.tabAt(event.pos())
-        self.dock_tabbar.setCurrentIndex(self.from_index)
-
-        if event.button() == Qt.RightButton:
-            if self.from_index == -1:
-                self.show_nontab_menu(event)
-            else:
-                self.show_tab_menu(event)
-
-    def tab_moved(self, event):
-        """Method called when a tab from a QTabBar has been moved."""
-        # If the left button isn't pressed anymore then return
-        if not event.buttons() & Qt.LeftButton:
-            self.to_index = None
-            return
-
-        self.to_index = self.dock_tabbar.tabAt(event.pos())
-
-        if not self.moving and self.from_index != -1 and self.to_index != -1:
-            QApplication.setOverrideCursor(Qt.ClosedHandCursor)
-            self.moving = True
-
-        if self.to_index == -1:
-            self.to_index = self.from_index
-
-        from_index, to_index = self.from_index, self.to_index
-        if from_index != to_index and from_index != -1 and to_index != -1:
-            self.move_tab(from_index, to_index)
-            self._fix_cursor(from_index, to_index)
-            self.from_index = to_index
-
-    def tab_released(self, event):
-        """Method called when a tab from a QTabBar has been released."""
-        QApplication.restoreOverrideCursor()
-        self.moving = False
-
-    def move_tab(self, from_index, to_index):
-        """Move a tab from a given index to a given index position."""
-        plugins = self._get_plugins()
-        from_plugin = self._get_plugin(from_index)
-        to_plugin = self._get_plugin(to_index)
-
-        from_idx = plugins.index(from_plugin)
-        to_idx = plugins.index(to_plugin)
-
-        plugins[from_idx], plugins[to_idx] = plugins[to_idx], plugins[from_idx]
-
-        for i in range(len(plugins)-1):
-            self.main.tabify_plugins(plugins[i], plugins[i+1])
-        from_plugin.dockwidget.raise_()
-
-    def show_tab_menu(self, event):
-        """Show the context menu assigned to tabs."""
-        self.show_nontab_menu(event)
+    Public interface for Spyder plugins.
 
-    def show_nontab_menu(self, event):
-        """Show the context menu assigned to nontabs section."""
-        menu = self.main.createPopupMenu()
-        menu.exec_(self.dock_tabbar.mapToGlobal(event.pos()))
-
-
-class SpyderDockWidget(QDockWidget):
```

Comment:
Don't worry, I moved this code to `widgets/dock.py` because I don't think we need to expose it as part of our API :-)